src/OrleansRuntime/Core/ManagementGrain.cs

```diff
@@ -310,5 +309,71 @@ private ISiloControl GetSiloControlReference(SiloAddress silo)
         {
             return InsideRuntimeClient.Current.InternalGrainFactory.GetSystemTarget<ISiloControl>(Constants.SiloControlId, silo);
         }
+
+        #region MultiCluster
+
+        public Task<List<IMultiClusterGatewayInfo>> GetMultiClusterGateways()
+        {
+            if (!Silo.CurrentSilo.GlobalConfig.HasMultiClusterNetwork)
+                throw new OrleansException("No multicluster network configured");
+
+            var multiClusterOracle = Silo.CurrentSilo.LocalMultiClusterOracle;
+            return Task.FromResult(multiClusterOracle.GetGateways().Cast<IMultiClusterGatewayInfo>().ToList());
+        }
+
+        public Task<MultiClusterConfiguration> GetMultiClusterConfiguration()
+        {
+            if (!Silo.CurrentSilo.GlobalConfig.HasMultiClusterNetwork)
+                throw new OrleansException("No multicluster network configured");
+
+            var multiClusterOracle = Silo.CurrentSilo.LocalMultiClusterOracle;
+            return Task.FromResult(multiClusterOracle.GetMultiClusterConfiguration());
+        }
+
+        public async Task<MultiClusterConfiguration> InjectMultiClusterConfiguration(IEnumerable<string> clusters, string comment = "", bool checkStabilityFirst = true)
+        {
+            if (!Silo.CurrentSilo.GlobalConfig.HasMultiClusterNetwork) 
+                throw new OrleansException("No multicluster network configured");
+
+            var multiClusterOracle = Silo.CurrentSilo.LocalMultiClusterOracle;
+
+            var configuration = new MultiClusterConfiguration(DateTime.UtcNow, clusters.ToList(), comment);
+
+            if (!MultiClusterConfiguration.OlderThan(multiClusterOracle.GetMultiClusterConfiguration(), configuration))
+                throw new OrleansException("Could not inject multi-cluster configuration: current configuration is newer than clock");
+
+            if (checkStabilityFirst)
+            {
+                try
+                {
+                    var unstableSilos = await multiClusterOracle.CheckMultiClusterStability(multiClusterOracle.GetMultiClusterConfiguration());
+
+                    if (unstableSilos.Count > 0)
+                    {
+                        var msg = string.Format("Found unstable silos {0}", string.Join(",", unstableSilos.Keys));
```

Comment:
As I already commented in another place, the notion of "unstable silos" is likely to cause confusion due to a potentially bad connotation with stability of the system.